model/host/db.go

```diff
@@ -128,6 +134,46 @@ func AllIdleEphemeral() ([]Host, error) {
 	return Find(query)
 }
 
+func IdleEphemeralGroupedByDistroId(creationTimeAsc bool) ([]HostsByDistro, error) {
+	var hostsByDistro []HostsByDistro
+	sortOrder := -1
+	if creationTimeAsc {
+		sortOrder = 1
+	}
+
+	pipeline := []bson.M{
+		{
+			"$match": bson.M{
+				RunningTaskKey:   bson.M{"$exists": false},
+				StartedByKey:     evergreen.User,
+				StatusKey:        evergreen.HostRunning,
+				ProviderKey:      bson.M{"$in": evergreen.ProviderSpawnable},
+				HasContainersKey: bson.M{"$ne": true},
+			},
+		},
+		{
+			"$sort": bson.M{CreateTimeKey: sortOrder},
+		},
+		{
+			"$group": bson.M{
+				"_id": "$" + bsonutil.GetDottedKeyName(DistroKey, DistroIDKey),
+				HostsByDistroHostsKey: bson.M{
+					"$push": "$$ROOT",
+				},
+			},
+		},
+		{
+			"$project": bson.M{"_id": 0, HostsByDistroDistroIDKey: "$_id", HostsByDistroHostsKey: 1},
+		},
+	}
+	err := db.Aggregate(Collection, pipeline, &hostsByDistro)
```

Comment:
nit: you can combine this line and the one below with the idiom `if err := func(); err != nil`

Reply:
Done.